### ISSUE TITLE
Embedding weight gradient comes back rank-4 and cannot be bound as a program input

### ISSUE THREAD

**Issue by pilkicTT** [tenstorrent/tt-mlir#9313]
`ttnn::embedding_bw` returns its result as `(1, 1, num_embeddings, embedding_dim)` whatever the rank of the weight it was given, while `ttnn.embedding_bw` declares the weight's own 2D shape. The tt-mlir runtime inserts the tensor as-is, so the gradient carries a rank-4 shape that disagrees with the program's declared output.

Ops downstream inside the same program tolerate the leading 1s. Binding the gradient as an input to the next program does not, which is what an optimizer step is:

```python
import torch

import tt_kurbla.torch


def main():
    weight = torch.randn(128, 64, dtype=torch.bfloat16, device="tt", requires_grad=True)
    indices = torch.randint(0, 16, (1, 32), dtype=torch.long)

    torch.nn.functional.embedding(indices.to("tt"), weight).backward(
        torch.randn(1, 32, 64, dtype=torch.bfloat16, device="tt")
    )
    print(f"grad shape -> {tuple(weight.grad.shape)}")
    (weight.grad + 1.0).cpu()


if __name__ == "__main__":
    main()
```

```
grad shape -> (128, 64)
TT_FATAL @ src/engine/execution_payload.cpp:61: actual.shape == expected.shape
bind_tensor: tensor for input 1 has the wrong shape. expected [128, 64]; got [1, 1, 128, 64]
```

Fix is in tt-mlir, `runtime/lib/ttnn/operations/embedding/embedding_backward.cpp`: reshape to the shape the op declares before `insertTTNNTensorAndValidate`, same as `pool/pool2d.cpp` already does for `avg_pool2d`. Needs an upstream PR and a submodule bump.

**Comment by pilkicTT** [comment]
> [!NOTE]
> This issue was automatically migrated from a private repository. Original author: @agobeljicTT
